specification/resourcehealth/resource-manager/Microsoft.ResourceHealth/preview/2021-01-01/ResourceHealth.json

```diff
@@ -1164,6 +1236,215 @@
         }
       }
     },
+    "potentialOutagesGetResult": {
+      "description": "The Get PotentialOutages operation response.",
+      "type": "object",
+      "$ref": "#/definitions/potentialoutage",
+      "readOnly": true
+    },
+    "potentialoutage": {
+      "type": "object",
+      "description": "Potential outages detected by Brain.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Id for the potential outage."
+        },
+        "title": {
+          "type": "string",
+          "description": "Title for the potential outage."
+        },
+        "confidenceScore": {
+          "type": "number",
+          "description": "Confidence score for the potential outage."
+        },
+        "confidence": {
+          "type": "string",
+          "description": "Confidence level for the potential outage."
+        },
+        "impactedRegions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Regions impacted by potential outages."
+        },
+        "detectedDateTime": {
+          "type": "string",
+          "description": "Time and date the potential outage is detected.",
+          "format": "date-time"
+        },
+        "impactedDataCenters": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Datacenters impacted by potential outages."
+        },
+        "impactedServices": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Services impacted by potential outages."
+        },
+        "impactedResourceTypes": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Resource types impacted by potential outages."
+        },
+        "symptoms": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Symptoms of potential outages."
+        },
+        "impactedTenants": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Tenants impacted by potential outages."
+        },
+        "impactedSubscriptions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "Subscriptions impacted by potential outages."
+        },
+        "impactedResources": {
```

Comment:
>impactedResources [](start = 9, length = 17)

This comment applies to all the array properties on this resource and any other resource. 

ARM has a response limit of 8MB for any request (LIST, GET, etc.) and there is also no paging support for lists inside a resource body. How are you planning to control the size of the responses if all these arrays are always included? What happens if a potential outage lists millions of resources and thousands of subs?

One option would be to truncate the contents to limit the size. Another would be to model the more populous arrays like impactedSubscriptions, impactedResources, impactedTenants, etc as a nested resource under the potentialOutage. This way they would be pageable via regular list calls.

Thoughts?

Reply:
those fields are taken out for now, and I would add them back as I get more insights as well as their use cases if it is considered reasonable.